src/main/java/address/Browser/BrowserManager.java

```diff
@@ -1,53 +1,177 @@
+                continue;
+            }
+            if (page.getPerson().equals(person)) {
+                indexOfSelectedTabBrowserPage = browser.indexOf(page);
+                break;
+            }
+        }
+        return indexOfSelectedTabBrowserPage;
     }
 
-    private void automateClickingAndScrolling() {
-        DOMElement container = browser.getDocument().findElement(By.id("js-pjax-container"));
-        DOMElement link = browser.getDocument().findElement(By.className("octicon octicon-repo"));
-        if (link != null) {
-            container.addEventListener(DOMEventType.OnLoad, e ->
+    private int findASuitableBrowserTabAndLoadThePersonGitHubProfilePage(Person person, List<Person> persons,
+                                                                         int indexOfContactsInFilteredListView) {
+
+        int indexOfSelectedTabBrowserPage = 0;
+
+        for (TabBrowserPage page : browser.getAllPages()) {
+            if (!page.isAssignedToLoadAPersonGitHubProfilePage()) {
+                continue;
+            }
+            int pagePersonIndex = persons.indexOf(page.getPerson());
+            if (pagePersonIndex < indexOfContactsInFilteredListView) {
+                indexOfContactsInFilteredListView = pagePersonIndex;
+                indexOfSelectedTabBrowserPage = browser.indexOf(page);
+            }
+        }
+        browser.getAllPages().get(indexOfSelectedTabBrowserPage).loadURL(person.getPersonGitHubProfilePageUrl(),
+                person);
+        return indexOfSelectedTabBrowserPage;
+    }
+
+    private void updateOtherBrowserTabs(List<Person> persons, int indexOfSelectedTabBrowserPage) {
+        int indexOfNextTabBrowserPage = (indexOfSelectedTabBrowserPage + 1) % browser.getNumberOfPages();
+        int indexOfNextPersonInFilteredList = (persons.indexOf(
+                                               browser.getPage(indexOfSelectedTabBrowserPage).getPerson()) + 1)
+                                               % persons.size();
+
+        do {
+            TabBrowserPage nextTabBrowserPage = browser.getPage(indexOfNextTabBrowserPage);
+            Person nextPersonInFilteredList = persons.get(indexOfNextPersonInFilteredList);
+            if (!nextTabBrowserPage.isAssignedToLoadAPersonGitHubProfilePage()
+                || !nextTabBrowserPage.getPerson().equals(nextPersonInFilteredList)) {
+                nextTabBrowserPage.loadURL(nextPersonInFilteredList.getPersonGitHubProfilePageUrl(),
+                                           nextPersonInFilteredList);
+            }
+            indexOfNextTabBrowserPage = (indexOfNextTabBrowserPage + 1) % browser.getNumberOfPages();
+            indexOfNextPersonInFilteredList = (indexOfNextPersonInFilteredList + 1) % persons.size();
+        } while (indexOfNextTabBrowserPage != indexOfSelectedTabBrowserPage);
+    }
+
+    private void automateClickingAndScrolling(Browser browser) {
```

Comment:
Good to add some comments providing some context (clicking what? scrolling where? when does it behave differently?)